File: sources/contract-registry/src/main/java/fr/unice/polytech/si5/al/e/contractRegistry/interfaces/ListContract.java
package fr.unice.polytech.si5.al.e.contractRegistry.interfaces;

import fr.unice.polytech.si5.al.e.contractRegistry.exceptions.NoSuchContractIdException;
import fr.unice.polytech.si5.al.e.model.Contract;
import fr.unice.polytech.si5.al.e.model.type.Type;

import javax.ejb.Local;
import java.util.Collection;

@Local
public interface ListContract {
    Collection<Contract> getContractByType(Type type);
    Contract getContractById(int id) throws NoSuchContractIdException;
}

File: blablamoveService/typescript/main/src/components/contract-registry/ListContract.ts
import { Type } from "../../entity/Type";
import { Contract } from "../../entity/contract/Contract";
import { Policy } from "../../entity/Policy/Policy";

export interface ListContract {
    getContractByType(type: Type): Promise<Array<Contract>>;
    getContractById(id: number): Promise<Contract>;
    getPolicyById(id: number): Promise<Policy>;
    getAllContract(): Promise<Array<Contract>>;
}
